src/app/hooks/useViewTracker/index.test.jsx

```diff
@@ -0,0 +1,182 @@
+  };
+  const { rerender } = renderHook(() => useViewTracker(data), { wrapper });
+
+  setIntersectionObserved();
+  rerender();
+
+  await act(() => wait(1100));
+
+  expect(useInView).toHaveBeenCalledWith({ threshold: 0.5 });
+  expect(spy).toHaveBeenCalledTimes(2);
+
+  const [componentViewEvent, pageViewEvent] = spy.mock.results;
+
+  expect(urlToObject(componentViewEvent.value)).toEqual({
+    origin: 'https://logws1363.ati-host.net',
+    pathname: '/',
+    searchParams: {
+      ati:
+        'PUB-[pidgin-mostRead]-[mostRead-most-read-view~view]-[]-[PAR=container-mostRead~CHD=link]-[news::pidgin.news.story.51745682.page]-[]-[]-[http://bbc.com/pidgin/tori-51745682]',
+      hl: '${timestamp}',
+      lng: '${browserLanguage}',
+      p: 'news::pidgin.news.story.51745682.page',
+      r: '${screenWidth}x${screenHeight}x${screenColorDepth}',
+      re: '${availableScreenWidth}x${availableScreenHeight}',
+      s: '598343',
+      s2: '70',
+      type: 'AT',
+    },
+  });
+
+  expect(urlToObject(pageViewEvent.value)).toEqual({
+    origin: 'https://logws1363.ati-host.net',
+    pathname: '/',
+    searchParams: {
+      ati:
+        'PUB-[pidgin-mostRead]-[mostRead-most-read-view~view]-[]-[PAR=container-mostRead~CHD=link]-[news::pidgin.news.story.51745682.page]-[]-[]-[http://bbc.com/pidgin/tori-51745682]',
+      hl: '${timestamp}',
+      lng: '${browserLanguage}',
+      p: 'news::pidgin.news.story.51745682.page',
+      r: '${screenWidth}x${screenHeight}x${screenColorDepth}',
+      re: '${availableScreenWidth}x${availableScreenHeight}',
+      s: '598343',
+      s2: '70',
+      type: 'AT',
+    },
+  });
+});
+
+it('should not call buildATIEventTrackUrl when element is in view for less than 1 second', async () => {
+  setIntersectionNotObserved();
+
+  const spy = jest.spyOn(atiUrl, 'buildATIEventTrackUrl');
+  const data = {
+    pageData,
+    componentName: 'mostRead',
+    actionLabel: 'most-read-view',
+  };
+
+  const { rerender } = renderHook(() => useViewTracker(data), { wrapper });
+
+  setIntersectionObserved();
+  rerender();
+
+  await act(() => wait(900));
+
+  expect(spy).not.toHaveBeenCalled();
+});
+
+it('should not call buildATIEventTrackUrl more than twice (once for component view and once for page view) when element is scrolled in and out of view', async () => {
+  setIntersectionNotObserved();
+
+  const spy = jest.spyOn(atiUrl, 'buildATIEventTrackUrl');
+  const data = {
+    pageData,
+    componentName: 'mostRead',
+    actionLabel: 'most-read-view',
+  };
+  const { rerender } = renderHook(() => useViewTracker(data), { wrapper });
+
+  // scroll element into view
+  setIntersectionObserved();
+  rerender();
+  await act(() => wait(1100));
+
+  // scroll element out of view
+  setIntersectionNotObserved();
+  rerender();
+
+  // scroll element into view again
+  setIntersectionObserved();
+  rerender();
+  await act(() => wait(1100));
+
+  expect(spy).toHaveBeenCalledTimes(2);
```

Comment:
Wasn't there a requirement to only despatch one view event per page load? Won't this send off two view events and skew the stats?

Reply:
There is that requirement and we've built it in. There are 2 calls made by sendEventBeacon - it sends another page view event as well as a component view event as shown here https://github.com/bbc/simorgh/blob/latest/src/app/containers/ATIAnalytics/beacon/index.js#L22-L23

Jon B mentioned that we do have to send 2 separate events but it seems weird to do this so I'll double check with him.

Reply:
Oh yeah, the current implementation sends a click and a view event, we discussed that needing to change. I'm interested to see how click tracking approach works with the view tracking; I do like that Web Core's implementation makes it impossible to send a click without a view.